google/auth/external_account_authorized_user.py

```diff
@@ -117,6 +115,10 @@ def __init__(
         self._client_secret = client_secret
         self._revoke_url = revoke_url
         self._quota_project_id = quota_project_id
+        self._scopes = scopes
+
+        if not self.valid and not self.can_refresh:
+            raise ValueError("Resulting credential can never be valid.")
```

Comment:
Error message isn't helpful, does not say why it is never valid. Can be improved like the old error

Reply:
Done